node/coinstacks/solana/api/src/websocket.ts

```diff
@@ -0,0 +1,87 @@
+import { Logger } from '@shapeshiftoss/logger'
+import { WebsocketClient as BaseWebsocketClient, Args, Options, Subscription } from '@shapeshiftoss/websocket'
+import WebSocket from 'ws'
+import { Transaction, WebsocketResponse } from './types'
+
+const logger = new Logger({ namespace: ['unchained', 'solana', 'websocket'], level: process.env.LOG_LEVEL })
+
+interface WebsocketArgs extends Omit<Args, 'logger'> {
+  transactionHandler: TransactionHandler | Array<TransactionHandler>
+}
+
+type TransactionHandler = (data: Transaction) => Promise<void>
+
+export class WebsocketClient extends BaseWebsocketClient {
```

Comment:
extending the `BaseWebsocketClient` abstract class. I know we talked about maybe not doing it right now, but just went ahead and updated real quick.

Reply:
Alright!